vid/src/avid_m/namespaced.rs

```diff
@@ -0,0 +1,241 @@
+//! This file implements the namespaced AvidM scheme.
+
+use super::{AvidMCommit, AvidMShare, RawAvidMShare};
+use crate::{
+    avid_m::{AvidMScheme, MerkleTree},
+    VidError, VidResult, VidScheme,
+};
+use jf_merkle_tree::MerkleTreeScheme;
+use serde::{Deserialize, Serialize};
+use std::ops::Range;
+
+/// Dummy struct for namespaced AvidM scheme
+pub struct NsAvidMScheme;
+
+/// Namespaced commitment type
+pub type NsAvidMCommit = super::AvidMCommit;
+/// Namespaced parameter type
+pub type NsAvidMParam = super::AvidMParam;
+
+/// Namespaced share for each storage node
+#[derive(Clone, Debug, Hash, Serialize, Deserialize, Eq, PartialEq, Default)]
+pub struct NsAvidMShare {
+    /// Index number of the given share.
+    index: u32,
+    /// The list of all namespace commitments
+    ns_commits: Vec<AvidMCommit>,
+    /// The size of each namespace
+    ns_lens: Vec<usize>,
+    /// Actual share content
+    content: Vec<RawAvidMShare>,
+}
+
+impl NsAvidMShare {
+    fn ns_share(&self, ns_id: usize) -> AvidMShare {
```

Comment:
nit: shall we call it `inner()`? to indicate the fact that it's a internal detail

Reply:
Done in 25c2965